test/error/error.cc

```diff
@@ -0,0 +1,184 @@
+#include <string>
+
+#include "gtest/gtest.h"
+
+#include "error/error.h"
+
+// Tests of GenericError.
+TEST(ErrorTest, ErrorMessage) {
+  GenericError e("abc");
+  EXPECT_EQ("abc", e.toString());
+}
+
+// Tests of ErrorOr.
+TEST(ErrorTest, ToStringValue) {
+  ErrorOr<std::string, GenericError> res("abc");
+  EXPECT_EQ("Ok", res.toString());
+}
+
+TEST(ErrorTest, ToStringError) {
+  ErrorOr<std::string, GenericError> res(GenericError("abc"));
+  EXPECT_EQ("abc", res.toString());
+}
+
+TEST(ErrorTest, ConstructorValueCopy) {
+  std::string a = "abc";
+  ErrorOr<std::string, GenericError> res(a);
+  EXPECT_TRUE(res.is_ok());
```

Comment:
ASSERT_TRUE would be more correct in this case as a failure of this will automatically cause a failure on the next check.
Same for the other tests.